Notification on Root Cause Analysis Returned

tinfernn opened the pull request:
# Feature Addition

CLOSES #254 

## Summary

Notifies supervisor (if assigned) and all admins when a root cause analysis is returned.

## Rationale

This keeps supervisors and admins up to date on changes made to incidents to ensure issues are dealt with in a timely manner.

## Design Documentation

N/A

## Changes

Creates a react() method in RootCauseAnalysisReturned.php to send a notification to the supervisor (if one is assigned) and to all admins when a root cause analysis is returned.

## Impact

Additional react() method to RootCauseAnalysisReturned.php.

## Testing

The following tests have been added to RootCauseAnalysisReturnedTest.php:
* test_returned_rca_notifies_supervisor_when_set()
* test_returned_rca_does_not_notify_supervisor_when_supervisor_is_not_set()
* test_returned_rca_notifies_admin_team()

## Screenshots/Video

![image](https://github.com/user-attachments/assets/774d30df-8d1c-40f5-b2f1-00953ab9cf6b)

## Checklist

- [x] Code follows the project's coding standards

- [x] Unit tests covering the new feature have been added

- [x] All existing tests pass

- [ ] The documentation has been updated to reflect the new feature

## Additional Notes

N/A

tinfernn committed:
Edited RootCauseAnalysisReturned.php to have a react() that sends a notification to supervisor (if assigned) and to admins when a root cause analysis has been returned.

Created tests in RootCauseAnalysisReturnedTest.php to test this feature.

welshy557 commented on app/Notifications/RootCauseAnalysis/RootCauseAnalysisReturnedNotification.php:
```diff
@@ -0,0 +1,45 @@
+        public string $incidentSlug,
+        public string $rootCauseAnalysisId,
+    ) {
+        $this->message = "The root cause analysis {$this->rootCauseAnalysisId} for incident #{$this->incidentSlug} has been reopened.";
```
Have the notification take in the admin user who returned and make the message the same as InvestigationReturned for consistency

welshy557 commented on app/StorableEvents/RootCauseAnalysis/RootCauseAnalysisReturned.php:
```diff
@@ -28,4 +31,17 @@ public function handle()
+        }
+
+        $admins = User::role('admin')->get();
+        Notification::send($admins, new RootCauseAnalysisReturnedNotification($incident->slug, $this->aggregateRootUuid()));
```
I don't think we need to notify admins, just the assigned supervisor will do. This isn't being done for investigations currently either. Also pull the supervisor from the rca instead of the incident as done in Investigation Returned as well.

welshy557 commented on resources/views/mail/root-cause-analysis-returned.blade.php:
```diff
@@ -0,0 +1,13 @@
+<x-mail::message>
+# Root Cause Analysis Returned
+
+A root cause analysis was returned.
```
Make it the same as the investigation returned email to stay consistent, which is just $this->message and a button to the rca.

tinfernn committed:
Merge branch 'staging' into notification-on-root-cause-analysis-returned

tinfernn committed:
Merge branch 'staging' into notification-on-root-cause-analysis-returned

tinfernn committed:
Edited constructor and react() method to pass in acting admin. Edited message sent as well.

Changed react() to not notify admins. Removed test for notifying admin team and not notifying supervisor. Edited other tests accordingly. Also edited two tests StatusTest.php to get them passing.

tinfernn committed:
Edited constructor and react() method to pass in acting admin. Edited message sent as well.

Changed react() to not notify admins. Removed test for notifying admin team and not notifying supervisor. Edited other tests accordingly. Also edited two tests StatusTest.php to get them passing.

Edited InvestigationReturned.php to use incident slug instead of id.
